tensorflow_probability/python/math/special.py

```diff
@@ -521,6 +545,332 @@ def betainc(a, b, x, name=None):
     return _betainc_custom_gradient(a, b, x)
 
 
+# The implementation of the inverse of the regularized incomplete beta function
+# is based on ideas and equations available in the following references:
+# [1] Milton Abramowitz and Irene A. Stegun
+#     Handbook of Mathematical Functions with Formulas, Graphs, and
+#         Mathematical Tables.
+#     US Government Printing Office, 1964 (reprinted 1972).
+#     https://archive.org/details/AandS-mono600
+# [2] William Press, Saul Teukolsky, William Vetterling and Brian Flannery
+#     Numerical Recipes: The Art of Scientific Computing
+#     Cambridge University Press, 2007 (Third Edition)
+#     http://numerical.recipes/book/book.html
+# [3] John Maddock, Paul A. Bristow, et al.
+#     The Incomplete Beta Function Inverses
+#     https://www.boost.org/doc/libs/1_79_0/libs/math/doc/html/special.html
+# [4] Stephen L. Moshier
+#     Cephes Mathematical Library
+#     https://netlib.org/cephes/
+
+
+def _betaincinv_initial_approx(a, b, y, dtype):
+  """Computes an initial approximation for `betaincinv(a, b, y)`."""
+  numpy_dtype = dtype_util.as_numpy_dtype(dtype)
+  tiny = tf.constant(np.finfo(numpy_dtype).tiny, dtype=dtype)
+  eps = tf.constant(np.finfo(numpy_dtype).eps, dtype=dtype)
+  one = tf.constant(1., dtype=dtype)
```

Comment:
FWIW can use `numpy_dtype` instead (just my own style preference, feel free to keep as is):

`one = tf.constant(1., dtype=dtype)` vs
`one = numpy_dtype(1.)`

Reply:
Done (here and elsewhere). Thanks for sharing this!